### PR TITLE
ci: test k8s version 1.31

### PR DESCRIPTION
**Pull request by blkperl**
### Motivation

Verifying the test suite passes on the new K8s release

### Modifications

* Changes CI to run against K8s 1.31

### Verification

E2E tests will suffice

### CODE REVIEW

File: .github/workflows/ci-build.yaml [hunk 1]
```diff
@@ -225,15 +225,15 @@ jobs:
             profile: minimal
             useApi: true
           - test: test-executor
-            install_k3s_version: v1.28.11+k3s1
+            install_k3s_version: v1.28.13+k3s1
```
Comment (blkperl) [inline .github/workflows/ci-build.yaml:228]: Note: We could change this to 1.29 as 1.28 is EOL in October.

File: docs/releases.md [hunk 2]
```diff
@@ -40,10 +40,10 @@ Otherwise, we typically release every two weeks:
...
 ## Kubernetes Compatibility Matrix
 
-| Argo Workflows \ Kubernetes | 1.28 | 1.29 | 1.30 |
-|-----------------------------|------|------|------|
-| **3.5**                     | `✓`  | `✓`  | `✓`  |
-| **3.4**                     | `?`  | `?`  | `?`  |
+| Argo Workflows \ Kubernetes | 1.28 | 1.29 | 1.30 | 1.31 |
+|-----------------------------|------|------|------|------|
+| **3.5**                     | `✓`  | `✓`  | `✓`  | `✓`  |
+| **3.4**                     | `?`  | `?`  | `?`  | `?`  |
```
Comment (agilgur5) [inline docs/releases.md:46]: This is still not quite correct per https://github.com/argoproj/argo-workflows/pull/13381#discussion_r1696170777

I'm thinking we just move `main` here into the table instead

Comment (blkperl) [reply docs/releases.md:47]: okay great, I added a `main` row

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**